### PR TITLE
Parameterized postsubmit arguments

### PR DESCRIPTION
**Pull request by micahhausler**
*Issue #, if available:*
N/A
*Description of changes:*

* Turned postsubmit args into parameterized flags
* Abstracted file changed variables 
* Added `--dry-run` flag

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/main_postsubmit.go [hunk 1]
```diff
@@ -1,128 +1,192 @@
...
+	dryRun bool
 }
 
-func (cmd Command) run(runMake bool, uploadArtifacts bool, targetPath string) {
-	if runMake == true {
-		// Add a specific target
-		cmd.args[1] = cmd.args[1] + targetPath
-		command := exec.Command("make", cmd.args...)
-		output, err := command.CombinedOutput()
+func (c *Command) runMake(uploadArtifacts bool, targetPath string) error {
```
Comment (bnrjee) [inline cmd/main_postsubmit.go:32]: This function tries to make 2 non make exec calls. Probably, we can rename it to only run?

File: cmd/main_postsubmit.go [hunk 2]
```diff
@@ -1,128 +1,192 @@
...
+	}
+	commandArgs = append(commandArgs, c.makeArgs...)
+
+	cmd := exec.Command("make", c.makeArgs...)
+	log.Printf("Executing: %s", strings.Join(cmd.Args, " "))
+	cmd.Stdout = outputStream
+	cmd.Stderr = errStream
+	if !c.dryRun {
+		err := cmd.Run()
 		if err != nil {
```
Comment (bnrjee) [inline cmd/main_postsubmit.go:46]: Probably we would want to print the output when this call succeeds. Same for the create_final_dir.sh script as well.

File: cmd/main_postsubmit.go (hunk 1 above)
Comment (micahhausler) [reply cmd/main_postsubmit.go:32]: yea, I can rename it

File: cmd/main_postsubmit.go (hunk 2 above)
Comment (micahhausler) [reply cmd/main_postsubmit.go:46]: The output of the command is set to forward to the tools stderr/stdout since we don't need to capture/parse it.

**Review by bnrjee — APPROVED** [review]

**Pull request merged**